azure/services/asogroups/spec.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package asogroups
+
+import (
+	"context"
+
+	asoresourcesv1 "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/external"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GroupSpec defines the specification for a Resource Group.
+type GroupSpec struct {
+	Client         client.Client // to fetch Azure(Managed)Cluster to set owner reference
```

Comment:
So far we've avoided adding any clients into the specs directly for async services. Instead, we usually use the client in `Reconcile()` to set any additional fields needed in the spec and then call `Parameters()` once they've been set. Is that something we want do here?

Reply:
That definitely seems reasonable. I suppose the way it is now is slightly more optimal in this case since we only use the client in the first reconciliation, where if that logic lived in `Reconcile`  and invoked before `CreateOrUpdateAsync`, then I think we'd have to do those GET calls with the client on every reconciliation.